### PR TITLE
Add pipeline for fetching and transforming submission portal entry

### PR DESCRIPTION
**Pull request by pkalita-lbl**
The main goal of these changes is to add two new Dagster jobs which support ingesting a single submission portal entry into Mongo. These two jobs are:

* `translate_metadata_submission_to_nmdc_schema_database`: This is intended as a sort of dry-run job. It fetches a submission from the portal API by the submission's UUID (provided to the job as run config), translates the data into an `nmdc:Database` object, and submits that data to the `/metadata/json:validate` endpoint. The `nmdc:Database` object is also serialized to JSON and stored in DRS for manual review.
* `ingest_metadata_submission`: This job does the same fetching and translation as the previous job. It submits the `nmdc:Database` object to the `metadata/json:submit` endpoint (which also does validation internally). That endpoint initiates a separate Dagster job so the `ingest_metadata_submission` will poll for the completion of the downstream job before it itself finishes. 

To support those jobs the following changes were made:

* Since the `/metadata` and `/runs` endpoints use user authentication (as opposed to site authentication) some bits of the existing `RuntimeApiSiteClient` class were extracted out into a new superclass (`RuntimeApiClient`). Then a new subclass, `RuntimeApiUserClient` was introduced. The new subclass was wrapped up as a Dagster resource.
* A `NmdcPortalApiClient` class was added to encapsulate interacting with the NMDC Data API (https://data-dev.microbiomedata.org/api/docs or https://data.microbiomedata.org/api/docs). This API currently requires authentication via a session cookie, which is a little odd. I believe that will be resolved in the not-too-distant future though. Once a cookie value is obtained it is passed as run config to either of the new jobs. 
* A `SubmissionPortalTranslator` class was added which handles translating data from the submission portal into an `nmdc:Database` instance. This class reuses parts of and follows the same general patterns of the previously introduced `GoldStudyTranslator` class.
* New Dagster ops were added for calling the relevant data access and translation methods.

Fixes https://github.com/microbiomedata/issues/issues/224

### CODE REVIEW

File: nmdc_runtime/site/graphs.py [hunk 1]
```diff
@@ -118,3 +120,12 @@ def gold_study_to_database():
 
     outputs = export_json_to_drs(database_dict, filename)
     add_output_run_event(outputs)
+
+@graph
+def metadata_submission_to_nmdc_schema_database():
+    metadata_submission = nmdc_portal_metadata_submission()
+    database = nmdc_schema_database_from_metadata_submission(metadata_submission)
```
Comment (pkalita-lbl) [inline nmdc_runtime/site/graphs.py:127]: Rename with verbs

Comment (dwinston) [reply nmdc_runtime/site/graphs.py:127]: nmdc_portal_metadata_submission -> fetch_portal_submission_by_id

**Comment by dwinston** [comment]
@pkalita-lbl I'm getting a testing error (see below). I'm not sure what's going on.

```
                actual = translator.get_database()
>               assert actual == expected
E               AssertionError: assert Database(acti..._version=None) == Database(acti..._version=None)
E                 
E                 Omitting 24 identical items, use -vv to show
E                 Differing attributes:
E                 ['study_set']
E                 
E                 Drill down into differing attribute study_set:
E                   study_set: [Study(id='fake:d862c2e3-6b0a-42f7-827c-67ebc8d44df7', name='Seasonal activities of the phyllosphere microbiome of perennial crops', description='Understanding the interactions between plants and microorganisms can inform microbiome management to enhance crop productivity and resilience to stress. Here, we apply a genome-centric approach to identify ecologically important leaf microbiome members on replicated plots of field-grown switc...
E                 
E                 ...Full output truncated (4 lines hidden), use '-vv' to show

tests/test_data/test_submission_portal_translator.py:252: AssertionError

```

**Comment by dwinston** [comment]
Perhaps it would help to convert the DBs to json? Perhaps pytest has native helpful json-diffing.

**Comment by pkalita-lbl** [comment]
Sorry about that! Will get it fixed up momentarily.

**Comment by dwinston** [comment]
I get another test failure now:

```
>           fetch_nmdc_portal_submission_by_id(
                op_context, "353d751f-cff0-4558-9051-25a87ba00d3f"
            )

tests/test_ops/test_data_api_ops.py:34: 
...
E           dagster._core.errors.DagsterInvalidConfigError: Error in config for op
E               Error 1: Missing required config entry "config" at the root. Sample config for missing entry: {'config': {'submission_id': '...'}}
```

It seems in `test_metadata_submission` you call `fetch_nmdc_portal_submission_by_id` directly with two positional args. I think you need to build the op context with the submission id config value before calling it.

File: nmdc_runtime/site/ops.py [hunk 2]
```diff
@@ -274,6 +287,41 @@ def create_objects_from_ops(context, op_docs: list):
     return op_docs
 
 
+@op(required_resource_keys={"runtime_api_user_client"})
+def validate_metadata(context: OpExecutionContext, database: nmdc.Database):
+    client: RuntimeApiUserClient = context.resources.runtime_api_user_client
+    response = client.validate_metadata(database)
+    response.raise_for_status()
```
Comment (dwinston) [inline nmdc_runtime/site/ops.py:294]: `response.raise_for_status()` is redundant now, as I added it to `RuntimeApiUserClient.request`. 🙂 Please remove here.

File: nmdc_runtime/site/ops.py [hunk 3]
```diff
@@ -274,6 +287,41 @@ def create_objects_from_ops(context, op_docs: list):
...
+            metadata={"detail": body["detail"]},
+        )
+    return body
+
+
+@op(required_resource_keys={"runtime_api_user_client"})
+def submit_metadata_to_db(context: OpExecutionContext, database: nmdc.Database) -> str:
+    client: RuntimeApiUserClient = context.resources.runtime_api_user_client
+    response = client.submit_metadata(database)
+    response.raise_for_status()
```
Comment (dwinston) [inline nmdc_runtime/site/ops.py:308]: `response.raise_for_status()` is redundant now, as I added it to `RuntimeApiUserClient.request`. 🙂 Please remove here.

File: nmdc_runtime/site/ops.py [hunk 4]
```diff
@@ -274,6 +287,41 @@ def create_objects_from_ops(context, op_docs: list):
...
+    body = response.json()
+    return body["detail"]["run_id"]
+
+
+@op(required_resource_keys={"runtime_api_user_client"})
+def poll_for_run_completion(context: OpExecutionContext, run_id: str) -> RunSummary:
+    client: RuntimeApiUserClient = context.resources.runtime_api_user_client
+    response = client.get_run_info(run_id)
+    response.raise_for_status()
```
Comment (dwinston) [inline nmdc_runtime/site/ops.py:317]: `response.raise_for_status()` is redundant now, as I added it to `RuntimeApiUserClient.request`. 🙂 Please remove here.

File: nmdc_runtime/site/ops.py [hunk 5]
```diff
@@ -591,9 +639,41 @@ def id_minter(*args, **kwargs):
...
+@op(required_resource_keys={"runtime_api_site_client"})
+def translate_portal_submission_to_nmdc_schema_database(
+    context: OpExecutionContext,
+    metadata_submission: Dict[str, Any],
+) -> nmdc.Database:
+    client: RuntimeApiSiteClient = context.resources.runtime_api_site_client
+
+    def id_minter(*args, **kwargs):
+        response = client.mint_id(*args, **kwargs)
+        response.raise_for_status()
```
Comment (dwinston) [inline nmdc_runtime/site/ops.py:661]: `response.raise_for_status()` is redundant now, as I added it to `RuntimeApiUserClient.request`. 🙂 Please remove here.

**Review by dwinston — CHANGES_REQUESTED** [review]
Generally good. Nice job abstracting out a `RuntimeApiClient`. The main thing is to refactor `validate_metadata` to not be a leaf op in its job, but then I think it's good to merge.

File: nmdc_runtime/site/graphs.py [hunk 6]
```diff
@@ -120,3 +125,24 @@ def gold_study_to_database():
...
     outputs = export_json_to_drs(database_dict, filename)
     add_output_run_event(outputs)
+
+
+@graph
+def translate_metadata_submission_to_nmdc_schema_database():
+    metadata_submission = fetch_nmdc_portal_submission_by_id()
+    database = translate_portal_submission_to_nmdc_schema_database(metadata_submission)
+
+    validate_metadata(database)
```
[the remark this answers is not in the document]
Comment (pkalita-lbl) [reply nmdc_runtime/site/graphs.py:135]: Yeah that was intentional. My thought was that even in the event that the translated `nmdc:Database` does not validate you might want to see what the pipeline produced as a debugging artifact. Is that an abuse of the DRS store?

[the remark this answers is not in the document]
Comment (dwinston) [reply nmdc_runtime/site/graphs.py:135]: No. It could be a bit more explicit in the graph flow, but it's fine.

**Pull request merged**